src/testing/mocks.ts

```diff
@@ -200,14 +200,12 @@ export function mockBuildCtx(config?: Config, compilerCtx?: CompilerCtx): BuildC
 }
 
 export function mockCache(config?: Config, compilerCtx?: CompilerCtx) {
-  if (!config) {
-    config = mockConfig();
-  }
+  const validatedConfig = mockValidatedConfig((config as Partial<ValidatedConfig>) ?? {});
```

Comment:
Is the type assertion here necessary? I would have thought TS woulda been happy about this, but maybe I'm missing something!

Reply:
yeah it is unfortunately, I thought that the typing of `config` as `d.Config` would satisfy `Partial<ValidatedConfig>` but it did not :/

I think we now have a few fields where the difference between them isn't just that they're optional in `Config` and required on `ValidatedConfig`